experimenter/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/FormFeatureValue/RichFeatureValueEditor.tsx

```diff
@@ -0,0 +1,435 @@
+  const { field } = useController({
+    control,
+    name: `${fieldNamePrefix}.value`,
+    defaultValue,
+  });
+
+  const hideSubmitErrors = useCallback(() => {
+    setSubmitErrors((prevSubmitErrors) => {
+      let newSubmitErrors = prevSubmitErrors;
+      if (prevSubmitErrors && prevSubmitErrors.value) {
+        newSubmitErrors = { ...prevSubmitErrors };
+        delete newSubmitErrors.value;
+      }
+      return newSubmitErrors;
+    });
+  }, [setSubmitErrors]);
+
+  const editor = useRef<ReactCodeMirrorRef>(null);
+
+  const schema = useMemo<Record<string, unknown> | null>(() => {
+    if (!featureConfig.schema) {
+      return null;
+    }
+
+    try {
+      return JSON.parse(featureConfig.schema);
+    } catch (e) {
+      return null;
+    }
+  }, [featureConfig.schema]);
+
+  const extensions = useMemo(() => {
+    const extensions = [
+      json(),
+      EditorView.focusChangeEffect.of(
+        (state: EditorState, focusing: boolean) => {
+          if (!focusing) {
+            field.onBlur();
+          }
+
+          return null;
+        },
+      ),
+      EditorView.updateListener.of((v: ViewUpdate) => {
+        if (v.docChanged) {
+          const value = v.state.doc.toString();
+          field.onChange(value);
+          hideSubmitErrors();
+        }
+      }),
+    ];
+
+    if (schema) {
+      extensions.push(linter(schemaLinter(schema)));
+
+      const completionSource = schemaAutocomplete(schema);
+      if (completionSource) {
+        extensions.push(
+          autocompletion({
+            override: [completionSource],
+          }),
+        );
+      }
+    }
+
+    return extensions;
+  }, [field, hideSubmitErrors, schema]);
+
+  // We can't use formControlAttrs() because this isn't actually a form component
+  // -- it is actually a bunch of divs, so there is no <input> to pass the return
+  // value of register() to.
+  //
+  // Instead we have to build up the class names ourself based on the logic in
+  // formControlAttrs.
+
+  return (
+    <ReactCodeMirror
+      value={defaultValue}
+      ref={editor}
+      extensions={extensions}
+      className={classNames("feature-value-editor", {
+        "is-valid":
+          !submitErrors?.value?.length && touched.value && !errors?.value,
+        "is-invalid":
+          submitErrors?.value?.length ||
+          ((touched.value ?? false) && (errors.value ?? false)),
+        "is-warning":
+          reviewWarnings.value?.length || reviewErrors.value?.length,
+      })}
+    />
+  );
+}
+
+/**
+ * A simplified schema that is generated by a feature manifest entry with only a
+ * simple list of variables.
+ */
+const simpleObjectSchema = z.object({
+  type: z.literal("object"),
+  properties: z.record(
+    z.object({
+      type: z.optional(
+        z.union([
+          z.literal("integer"),
+          z.literal("boolean"),
+          z.literal("string"),
+          z.literal("json"),
+        ]),
+      ),
+    }),
+  ),
+});
```

Comment:
Wait why can't we use the jsonschemas that are already present in the backend?

Reply:
This is the Zod definition of a JSON schema "object" type with simple properties